hazelcast/src/main/java/com/hazelcast/internal/adapter/DataStructureAdapter.java

```diff
@@ -56,4 +56,43 @@
     LocalMapStats getLocalMapStats();
 
     boolean containsKey(K key);
+
+    /**
+     * Contains all methods of {@link DataStructureAdapter}.
+     */
+    enum DataStructureMethods implements DataStructureAdapterMethod {
```

Comment:
NOTE: It's still not a constraint that all methods of `DataStructureAdapter` have to be defined in `DataStructureMethods`. I think that's okay, e.g. if they are anyway dependent on another method which is already defined.

If that check is needed it should be easy to add to the existing test (get all methods from `DataStructureAdapter` and check if they are in the `knownMethods` map of the test.